### PR TITLE
Fix calculating position and width for appointments with different cases of start & end (T819852)

### PR DESCRIPTION
**Pull request by Olexi**

### CODE REVIEW

File: js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.base.js [hunk 1]
```diff
@@ -499,6 +499,13 @@ class BaseRenderingStrategy {
             if(endDate > viewEndDate) {
                 endDate = viewEndDate;
             }
+
+            var currentDayViewEndHour = new Date(endDate);
```
Comment (Yarovaya) [inline js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.base.js]: Is it possible to rename?
currentDayViewEndHour -> currentViewEndTime or allowableEndTime

**Review by Yarovaya — CHANGES_REQUESTED** [review]

**Review by Yarovaya — DISMISSED** [review]

**Review by Yarovaya — DISMISSED** [review]

File: js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.base.js [hunk 2]
```diff
@@ -466,17 +466,15 @@ class BaseRenderingStrategy {
         return startDate;
     }
 
-    endDate(appointment, position, isRecurring) {
-        var endDate = this.instance._getEndDate(appointment),
+    endDate(appointment, position, isRecurring, ignoreViewDates = false) {
+        let endDate = this.instance._getEndDate(appointment, ignoreViewDates),
             realStartDate = this.startDate(appointment, true),
             viewStartDate = this.startDate(appointment, false, position);
```
Comment (Krijovnick) [inline js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.base.js:472]: realStartDate and viewStartDate are const
fullDuration is let

File: js/ui/scheduler/workspaces/ui.scheduler.work_space.js [hunk 3]
```diff
@@ -2152,10 +2152,14 @@ var SchedulerWorkSpace = Widget.inherit({
     },
 
     getTimeShift: function(date) {
-        var cellDuration = this.getCellDuration(),
+        let cellDuration = this.getCellDuration(),
```
Comment (Krijovnick) [inline js/ui/scheduler/workspaces/ui.scheduler.work_space.js]: const

File: js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.horizontal_month_line.js [hunk 4]
```diff
@@ -8,13 +8,13 @@ var HOURS_IN_DAY = 24,
...
-            endDate = new Date(this.endDate(appointment, position, isRecurring)),
+        let startDate = new Date(this.startDate(appointment, false, position)),
+            endDate = new Date(this.endDate(appointment, position, isRecurring, true)),
             cellWidth = this.getDefaultCellWidth() || this.getAppointmentMinSize();
 
         startDate = dateUtils.trimTime(startDate);
 
-        var width = Math.ceil(this._getDurationInHour(startDate, endDate) / HOURS_IN_DAY) * cellWidth;
+        let width = Math.ceil(this._getDurationInHour(startDate, endDate) / HOURS_IN_DAY) * cellWidth;
         width = this.cropAppointmentWidth(width, cellWidth);
```
Comment (Krijovnick) [inline js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.horizontal_month_line.js:17]: there is we can use const and rewrite 
```
const startDate = dateUtils.trimTime(new Date(this.startDate(appointment, false, position)))
```
and 
```
const duration = this._getDurationInHour(startDate, endDate) / HOURS_IN_DAY;
const width = this.cropAppointmentWidth(Math.ceil(duration * cellWidth, cellWidth);
```

File: js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.base.js (hunk 2 above)
Comment (Olexi) [reply js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.base.js:472]: fullDuration is changed in this method. So, it is needed as let

**Review by Krijovnick — APPROVED** [review]

**Review by Yarovaya — APPROVED** [review]

**Review by andrewmakarov — APPROVED** [review]

**Pull request merged**